### PR TITLE
feat(datetime):add time-compact presentation style

### LINKED ISSUE #24402: feat: Datetime add time-compact presentation style
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

When used with `time-date` presentation style or similar, the datetime component renders the time as follows:

image

When used as `time`, sadly like this:
image

### Describe the Use Case

- When multiple times are needed for only one date, ex. Start & end times

### Describe Preferred Solution

I suggest a new presentation style `time-compact`, that only renders the time picker as seen in `time-date` presentation.

### Describe Alternatives

_No response_

### Related Code

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by JesseB0rn**
Add presentation style 'time-compact', that shows
a time picker identical to the ones used in
presentation styles'time-date', 'date-time' or similar.
I need a start and end time for one of my projects,
and the normal timepicker looked a bit ridiculous.

resolves #24402

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #24402

## What is the new behavior?

- Adds the 'time-compact' presentation style, which is identical to the time in 'time-date'

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by sean-perkins** [comment]
Hello @JesseB0rn thanks for the PR!

It looks like a lot of the changes are a result of formatting, likely from your local configuration with an extension like Prettier. Can you update your PR to not have this formatting applied? If you are using VSCode, you can save with CMD + Shift + P and select "Save without Formatting". 

We have plans to institute global formatting rules in the future.

**Comment by JesseB0rn** [comment]
Reverted formatting in latest commit

File: core/src/components/datetime/usage/vue.md [hunk 1]
```diff
@@ -88,7 +99,7 @@
 </template>
 
 <script>
-  import { defineComponent, ref } from 'vue';
+  import { defineComponent, ref } from "vue";
```
Comment (sean-perkins) [inline core/src/components/datetime/usage/vue.md:102]: Can we revert code formatting with the double/single quotes in this file.

File: core/src/components/datetime/usage/javascript.md [hunk 2]
```diff
@@ -81,9 +92,9 @@
 ```
 
 ```javascript
-import { format, parseISO } from 'date-fns';
+import { format, parseISO } from "date-fns";
```
Comment (sean-perkins) [inline core/src/components/datetime/usage/javascript.md:95]: Can we revert code formatting with the double/single quotes in this file (just for the JavaScript section).

**Review by sean-perkins — CHANGES_REQUESTED** [review]
Few small formatting tweaks remain.

Can you also add the `time-compact` example to our test file: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/datetime/test/presentation/index.html

You can just add this mark-up after one of the existing test cases.

```html
<div class="grid-item">
  <h2>time-compact</h2>
  <ion-datetime
    presentation="time-compact"
  ></ion-datetime>
</div>
```

After that we are ready to run CI 🏃

**Comment by JesseB0rn** [comment]
What is the state on this?

**Comment by sean-perkins** [comment]
Hello @JesseB0rn the team is just getting back from Holiday break. We have a bit of a backlog if issues and open items to review; prioritizing bugs over features at the moment. 

I'll run the workflow to make sure all checks pass and then discuss with the team when this change could make sense to be merged. Since it's a feature, it'd likely be 6.1+.

**Comment by sean-perkins** [comment]
Hello @JesseB0rn ran this locally and things look good, nice work!

There appears to be a different bug where in the `display.html` example test, on first changing the presentation value, the popover for the time-compact won't display until toggling to cover and back. This is indifferent to your changes, but wanted to mention it.

Can you rework the commit message to have a space after the colon `:`?
> feat(datetime): add time-compact presentation style

Since this is a feature, we will need to wait until we are ready to move to 6.1.

**Comment by JesseB0rn** [comment]
> Can you rework the commit message to have a space after the colon :?

I made the mistake of already pushing the commit here, I can't get git to let me rebase the commit for some reason, sorry.

**Comment by amandaesmith3** [comment]
Closing per https://github.com/ionic-team/ionic-framework/issues/24402#issuecomment-1050034535.

**Pull request closed without merging**